### PR TITLE
Added malloc checks

### PR DESCRIPTION
**Pull request by ericwol-msft**

### CODE REVIEW

**Review by vaavva — APPROVED** [review]

File: src/uhttp.c [hunk 1]
```diff
@@ -231,12 +234,17 @@ static int process_header_line(const unsigned char* buffer, size_t len, size_t*
...
-                char* headerValue = (char*)malloc(valueLen+1);
-                if (headerValue == NULL)
+                size_t valueLen = safe_subtract_size_t((&buffer[index]), targetPos);
+                size_t malloc_size = safe_add_size_t(valueLen, 1);
+
+                char* headerValue;
+                if (malloc_size == SIZE_MAX ||
+                    (headerValue = (char*)malloc(malloc_size)) == NULL)
                 {
+                    LogError("Cannot malloc memory, size:%zu", malloc_size);
```
Comment (ewertons) [inline src/uhttp.c:244]: Could the message be a little more specific please, for systems where the files and line numbers are not printed?

File: src/uhttp.c [hunk 2]
```diff
@@ -202,9 +203,11 @@ static int process_header_line(const unsigned char* buffer, size_t len, size_t*
                     free(headerKey);
                     headerKey = NULL;
                 }
-                headerKey = (char*)malloc(keyLen + 1);
-                if (headerKey == NULL)
+                size_t malloc_size = safe_add_size_t(keyLen, 1);
+                if (malloc_size == SIZE_MAX ||
+                    (headerKey = (char*)malloc(malloc_size)) == NULL)
                 {
+                    LogError("Cannot malloc memory, size:%zu", malloc_size);
```
Comment (ewertons) [inline src/uhttp.c:210]: Could the message be a little more specific please, for systems where the files and line numbers are not printed?

```suggestion
                    LogError("Cannot malloc headerKey, size:%zu", malloc_size);
```

File: src/uhttp.c (hunk 1 above)
Comment (ewertons) [reply src/uhttp.c:244]:
```suggestion
                    LogError("Cannot malloc headerValue, size:%zu", malloc_size);
```

**Review by ewertons — APPROVED** [review]
Approved with suggestions.

**Pull request merged**